tests/phpunit/CRM/Contribute/Form/Contribution/ConfirmTest.php

```diff
@@ -102,7 +102,7 @@ public function testPayNowPayment(): void {
     );
 
     // Make sure that certain parameters are set on return from processConfirm
-    $this->assertEquals('Campaign Contribution', CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'financial_type_id', $processConfirmResult['financial_type_id']));
+    $this->assertEquals('Campaign Contribution', CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'financial_type_id', $processConfirmResult['contribution']->financial_type_id));
```

Comment:
I'm a bit unsure about this one - this signals that nor returning financial_type_id in result is impacting on what the processConfirm function is returning to the functions that call it - in essence a hidden merge of $paymentParams into what was being returned was going on.

That is
1) bad
2) possibly risky to change

Rather than altering the test we could alter the lines that return the $result such that they merge in the $paymentParams perhaps

Reply:
@eileenmcnaughton doPayment is not supposed to be returning `financial_type_id` (see https://lab.civicrm.org/dev/financial/-/issues/141).

The test is a bit messy because it's merging stuff together and then checking that something exists in the array but what it really wants to do is check that the contribution has the right financial type - which it is now doing because there's a copy of the contribution in the result. Happy to see the test cleaned up but feel like it's out of scope for the change to the dummy payment processor here?

Reply:
@mattwire I'm not talking about cleaning up the test - I'm suggesting we change the PR so there is no change to what is returned from the funciton being tested & hence the test is unchanged. The alternative is to do more analysis on how the results of that function are used to be sure that there is no impact from it changing here

Reply:
@eileenmcnaughton ok try now